smithy-aws-cloudformation-traits/src/main/java/software/amazon/smithy/aws/cloudformation/traits/CfnRootResourceIdTraitValidator.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.aws.cloudformation.traits;
+
+import java.util.ArrayList;
+import java.util.List;
+import static software.amazon.smithy.model.validation.ValidationUtils.splitCamelCaseWord;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.validation.AbstractValidator;
+import software.amazon.smithy.model.validation.ValidationEvent;
+
+
+
+
+/**
+ * Validates that members marked as having @CfnRootResourceId are marked within the listAPI
+ * and not in other APIs.
+ */
+public final class CfnRootResourceIdTraitValidator extends AbstractValidator {
```

Comment:
This should have tests. We generally test validators using "errorfiles" -- create a foo.smithy Smithy file that has issues in it, and an accompanying foo.errors that contains each error you expect (removing the filename and location). You can see a bunch of example here: https://github.com/awslabs/smithy/tree/main/smithy-aws-cloudformation-traits/src/test/resources/software/amazon/smithy/aws/cloudformation/traits/errorfiles

Reply:
Resolving as we are not moving forward with cfnRootResourceId. With discussions with David, we can use resource identifier bindings as well as knowledge index to achieve the behavior we want.